sql/hive/src/main/scala/org/apache/spark/sql/hive/client/IsolatedClientLoader.scala

```diff
@@ -191,8 +187,8 @@ private[hive] class IsolatedClientLoader(
     (sharesHadoopClasses && isHadoopClass) ||
     name.startsWith("scala.") ||
     (name.startsWith("com.google") && !name.startsWith("com.google.cloud")) ||
-    name.startsWith("java.lang.") ||
-    name.startsWith("java.net") ||
+    name.startsWith("java.") ||
+    name.startsWith("javax.") ||
```

Comment:
`javax.sql` et al were not found without this change. I think they should always have been considered non-shared classes? they're in the JDK from Java 8 at least.

Reply:
yes I found the same, this seems like a mistake.  I also noticed that I needed to add at least

```
    // needed for at least com.sun.security.auth.module.UnixLoginModule
    name.startsWith("com.sun") ||
```

When I ran some tests w/ security.  I wondered whether I should make that rule even tighter, but I figured everything under com.sun is probably considered shared.

Reply:
Would it make sense to just do `javax.sql` in the starts with here given other things like `javax.servlet` or are those non-shareable as well?